### ISSUE TITLE
Sol Trader still remembers Syndicate code words from a past life

### ISSUE THREAD

**Issue by marcjethro** [ParadiseSS13/Paradise#23536]
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

515.1620

### Issue Description

I was a syndicate that died. Respawned as a Sol Trader, but the Syndicate code words were still highlighted.
![Screenshot 2023-12-13 010335](https://github.com/ParadiseSS13/Paradise/assets/78044467/6e1f754d-3133-48f9-8f18-387eed6bfb7c)

### What did you expect to happen?

Syndicate code words wouldn't be highlighted anymore

### What happened instead?

Syndicate code words were highlighted

### Why is this bad/What are the consequences?

It doesn't affect much I guess, it's just weird

### Steps to reproduce the issue.

Be traitor
Die
Respawn as sol trader
Say the syndicate code words
See that it's highlighted

### When did the problem start happening?

Didn't notice it before

**Comment by Contrabang** [comment]
Yep, this because you technically inhabit the same mind, the same kind of stuff occurs for some other midrounds too, like revenants, morphs, slaugther demons, etc. Thank you for the report!
